### PR TITLE
Implemented a User_Input_Parser class to validate user's setting files

### PR DESCRIPTION
**Pull request by jkalish14**
Created a User_Input_Parser class to inspect a user's provided settings. If the arguments are of the wrong type, or do not meet certain criteria (as set in the utils/utils.py  default_*_settings dictionaries), the user will receive an INFO message and the default value will be used.

Also updated the .gitignore to include VSCode files and the default files that are created in the project structure when "blankly init" is run from the command line

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -142,4 +142,10 @@ notify.json
 .pytype/
 
 # Cython debug symbols
-cython_debug/
\ No newline at end of file
+cython_debug/
+
+# User-Specific Files
+*.json
+*.code-workspace
```
Comment (EmersonDove) [inline .gitignore:149]: Code workspace is a good add but generally we want to keep .json, requirements and bot.py

File: blankly/utils/utils.py [hunk 2]
```diff
@@ -33,60 +34,282 @@
...
+    """
+    # Regular Expression soruce https://www.geeksforgeeks.org/check-if-email-address-valid-or-not-in-python/
+    REG_EX = r'\b[A-Za-z0-9._%+-]+@[A-Za-z0-9.-]+\.[A-Z|a-z]{2,}\b'
+
+    try:
+        re.fullmatch(REG_EX, val)
+    except:
+        return False
+
+    return True
```
Comment (EmersonDove) [inline blankly/utils/utils.py:172]: this is pretty cool, great implementation

File: blankly/utils/utils.py [hunk 3]
```diff
@@ -126,15 +349,30 @@ def __compare_dicts(self, default_settings, user_settings):
                     warning_string = "\"" + str(k) + "\" not specified in preferences, defaulting to: \"" + str(v) + \
                                      "\""
                     info_print(warning_string)
-                    user_settings[k] = v
+                    user_settings[k] = v.default
```
Comment (jkalish14) [inline blankly/utils/utils.py:352]: noticed this mistake earlier, thought corrected it prior to pushing. At this point, v is still a dict, so it has no method "default". Will correct and push the changes

**Comment by jkalish14** [comment]
Added user-defined benchmark to the backtest settings. Adding a symbol (for stocks or crypto) will add a trace to the last plot of the backtest results which shows the portfolio performance if it was entirely invested into the benchmark. Implementation avoids the benchmark counting towards the account value. Additionally, specifying a benchmark will add beta to the metrics. 

Note, there does seem to be an issue in the beta calculation because a backtest that buys and holds SPY which also uses SPY as a benchmark does not result in a beta of 1. Not too sure what's going on, but i will continue to look into it.

**Pull request closed without merging**